examples/LoggingExtensions/Program.cs

```diff
@@ -16,19 +16,16 @@
 
 using System.Diagnostics.Tracing;
 using Examples.LoggingExtensions;
+using OpenTelemetry;
 using OpenTelemetry.Logs;
 using OpenTelemetry.Resources;
 using Serilog;
 
-var resourceBuilder = ResourceBuilder.CreateDefault().AddService("Examples.LoggingExtensions");
-
-var openTelemetryLoggerProvider = new OpenTelemetryLoggerProvider(options =>
-{
-    options.IncludeFormattedMessage = true;
-    options
-        .SetResourceBuilder(resourceBuilder)
-        .AddConsoleExporter();
-});
+var openTelemetryLoggerProvider = Sdk.CreateLoggerProviderBuilder(
+    options => options.IncludeFormattedMessage = true)
+    .ConfigureResource(builder => builder.AddService("Examples.LoggingExtensions"))
+    .AddConsoleExporter()
+    .Build();
```

Comment:
Feels a little off to have some things configured via the options like `IncludeFormattedMessage` and other things configured via methods off the builder.

For metrics we have some similar options like max streams that we set using builder methods:

```csharp
Sdk.CreateMeterProviderBuilder()
   .SetMaxMetricStreams(100);
```

This PR is growing, so maybe a follow up? Though a follow up would mean the potential for some breaking changes.

Reply:
Ya this was something I went back and forth on. The bools suck because you can't chain them which kind of breaks the builder pattern. That's why I added the delegate. We could add "Set" methods to preserve the chaining but some downsides to that: More public API and maybe confusing to have properties with get/set and also "Set" methods?

@utpilla @cijothomas @reyang Thoughts?

Reply:
I guess it all started from [`SetSampler`](https://github.com/open-telemetry/opentelemetry-dotnet/blob/27658f058cc248500805d66ae2ab65c2099d6400/docs/trace/extending-the-sdk/Program.cs#L31).

I personally would find it slightly more user-friendly with `builder.SetSomething` if the number of things to be set is small enough, and the things to be set are fairly complex (e.g. very hard to be configuration-based). I would be less happy if there are too many simple options that I have to chain the builder calls:

```csharp
// it is just more difficult if I want to implement a configuration file

Sdk.CreateMeterProviderBuilder()
   .SetMaxMetricStreams(100),
   .SetX(true),
   .SetY("abc"),
   .SetZ(false);
```

Reply:
Btw, now that we started talking about configuration, do you think `ConfigureResource` should be `SetResource`? 😆

Reply:
@reyang You are going to love this, we actually have `SetResourceBuilder` & `ConfigureResource` both today 🤣 [ConfigureResource was a recent addition](https://github.com/open-telemetry/opentelemetry-dotnet/pull/3307).

Subtle differences between the two. `Set` clobbers anything that was configured. `Configure` is additive. For users configuring their host, I would expect them to call `Set` (but they could also `Configure`). For libraries/extension authors, they _should_ call `Configure`.

We should probably obsolete the `Set` version because in `Configure` you can call `Clear` on the builder if you really wanted to reset it. (Pretty sure, didn't double-check a `Clear` exists.)

Reply:
I would vote for eventually getting rid of `ConfigureXyz` because "configure" is a very general term that can have several different meanings (e.g. it could be additive and commutative; it could be additive but not commutative; it could be idempotent...).

If we're trying to be pedantic, "Append" is probably good for Processor and Exporter (coz it is not commutative due to the sequential processing behavior), "Set" is probably good for Sampler, "Configure" is probably good for anything that takes a complex "XyzOptions" object.

Reply:
I removed the delegate and replaced it with 3 new "Set" methods. Don't disagree with any of this, it just seems like a bigger conversation we need to have beyond this PR.

Reply:
+1, I would suggest that we scope this out for now. Maybe revisit it after this PR.